yaspin/core.py

```diff
@@ -263,6 +265,19 @@ def hide(self):
                 # can be rewritten to
                 sys.stdout.flush()
 
+    @contextlib.contextmanager
+    def hidden(self):
+        """Hide the spinner within a block, can be nested"""
+        if self._hidden_level == 0:
+            self.hide()
+        self._hidden_level += 1
+
+        yield
```

Comment:
Using such contextmanager as follows: `with sp.hidden() as foo:` gives `foo` of type `None`. I'm not sure whether it's a good or bad decision. On the one hand, it avoids confusion with `sp` variable (which is already produced by another contextmanager and returns `self`). On the other, it prohibits a pretty popular usage pattern for contextamanager. What do you think?

Reply:
The user already has the spinner object (as they are calling `.hidden()` on it), so yielding it again would not add any value for the user. I always regarded the object we yield as something related to the context of the with, but with `.hidden()` all context for the block is kept within the Spinner class itself. So if we can't supply anything useful I'd say we don't duplicate anything and go for yielding nothing.

Reply:
Totally agree